File: getter_setter.js
class Pokemon {
  constructor(id, name, type) {
    this.id = id;
    this.name = name;
    this.type = type;
  }
  get id() {
    return this.id
  }
  set id(value) {
    if(Number.isInteger(value) === true) {
      this._id = value;
    } else {
      console.log("id 타입은 정수여야 합니다 ")
    }
  }
}

const pikachu = new Pokemon(1.2, "피카츄", "전기")
console.log(pikachu);

File: getter_setter.py
# from msilib import type_binary


class Pokemon:
  def __init__(self,id,name,type):
    self.id = id
    self.name = name
    self.type = type
  @property
  def id(self):
    return self.__id
  
  @id.setter
  def id(self, id):
      if str(type(id)) == "<class 'int'>":
        self.__id = id
      else:
        print("id타입은 정수여야 합니다.")

pikachu = Pokemon(3.3, "피카츄", "전기")
print(pikachu.__dict__)